### PR TITLE
Dependabot Docs: update guidance-for-the-configuration-of-private-registries

### PR DESCRIPTION
**Pull request by whatatripp**
### Why:

Added guidance for configuring ProGet as a private registry; this was requested be a few mutual customers, as they weren't sure what fields to enter as the token, etc.

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Just adding an example for ProGet that's formatted just like the others.

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline.

- [X] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

File: content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md [hunk 1]
```diff
@@ -627,3 +628,25 @@ If you are restricting which IPs can reach your Nexus host, you need to add the
...
+```yaml
+registries:
+  proget-nuget-feed:
+    type: nuget-feed
+    url: https://proget.corp.local/nuget/MyNuGetFeed/v3/index.json
+    token: ${{secrets.PROGET_APK_KEY}}
+```
+
+{% endraw %}
+
```
Comment (mchammer01) [inline content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md:649]: Let's follow the same structure as for the other private registry hosts.
```suggestion

#### Notes

```

File: content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md [hunk 2]
```diff
@@ -627,3 +628,25 @@ If you are restricting which IPs can reach your Nexus host, you need to add the
...
+The `token` should be an API Key that with access to view packages; see [API Access and API Keys](https://docs.inedo.com/docs/buildmaster-administration-security-api-keys).
```
Comment (mchammer01) [inline content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md:650]:
```suggestion
The `token` should be an API Key with access to view packages. For more information, see [API Access and API Keys](https://docs.inedo.com/docs/buildmaster-administration-security-api-keys) in the ProGet documentation.
```

File: content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md [hunk 3]
```diff
@@ -627,3 +628,25 @@ If you are restricting which IPs can reach your Nexus host, you need to add the
 For more information, see [Securing Nexus Repository Manager](https://help.sonatype.com/repomanager3/planning-your-implementation/securing-nexus-repository-manager) in the Sonatype documentation.
 
   Registries can be proxied to reach out to a public registry in case a dependency is not available in the private regstry.{% ifversion dependabot-private-registries %} However, you may want {% data variables.product.prodname_dependabot %} to only access the private registry and not access the public regsitry at all. For more information, see [Quick Start Guide - Proxying Maven and NPM](https://help.sonatype.com/repomanager3/planning-your-implementation/quick-start-guide---proxying-maven-and-npm)" in the Sonatype documentation, and "[AUTOTITLE](/code-security/dependabot/working-with-dependabot/removing-dependabot-access-to-public-registries)."{% endif %}
+
+### ProGet
+
+For information about ProGet and instructions on how to configure {% data variables.product.prodname_dependabot %} to work with feeds in ProGet, see [the ProGet documentation](https://docs.inedo.com/docs/proget-overview).
```
Comment (mchammer01) [inline content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md:634]: Minor
```suggestion
For information about ProGet and instructions on how to configure {% data variables.product.prodname_dependabot %} to work with feeds in ProGet, see the [ProGet documentation](https://docs.inedo.com/docs/proget-overview).
```

**Review by mchammer01 — COMMENTED** [review]
@whatatripp - thanks for adding another private repository host.
I've made a few editorial suggestions that I'm going to apply, and then I'll ask one of the relevant team for a technical review.

**Comment by mchammer01** [comment]
I've asked for a technical review about the Bundler and Python support for ProGet as I am not sure this is correct 🤔

**Comment by mchammer01** [comment]
Just found this: https://docs.inedo.com/docs/proget-feeds-pypi and this: https://docs.inedo.com/docs/proget-feeds-rubygem

File: content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md [hunk 4]
```diff
@@ -39,12 +39,13 @@ You'll find detailed guidance for the setup of the following package managers an
...
 - [{% data variables.product.prodname_registry %} registry](#data-variablesproductprodname_registry--registry)
 - [Nexus](#nexus)
+- [ProGet](#proget)
 
 ## Configuring package managers
 
 ### Bundler
 
-Supported by Artifactory, Artifacts, {% data variables.product.prodname_registry %} registry, and Nexus.
+Supported by Artifactory, Artifacts, {% data variables.product.prodname_registry %} registry, Nexus, and ProGet.
```
Comment (jurre) [inline content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md:48]: @whatatripp it doesn't seem like ProGet supports Bundler from reading the docs right?

```suggestion
```

File: content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md [hunk 5]
```diff
@@ -351,7 +352,7 @@ registries:
 
 ### Python
 
-Supported by Artifactory, Azure Artifacts, and Nexus. The {% data variables.product.prodname_registry %} registry is not supported.
+Supported by Artifactory, Azure Artifacts, Nexus, and ProGet. The {% data variables.product.prodname_registry %} registry is not supported.
```
Comment (jurre) [inline content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md:355]: @whatatripp it doesn't seem like ProGet supports Python packages from reading the docs right?

```suggestion
```

**Comment by jurre** [comment]
> Just found this: https://docs.inedo.com/docs/proget-feeds-pypi and this: https://docs.inedo.com/docs/proget-feeds-rubygem

Ah, nice find! Maybe let's add examples of the config file entries required to set that up in that case?

**Comment by mchammer01** [comment]
@whatatripp 👋🏻 - I agree with @jurre here. If these ecosystems/languages are supported, it would be great to have specific examples in the docs, as the one you provided for NuGet. Can you provide us with an example for Python, and another one for Bundler? 🙏🏻

**Comment by whatatripp** [comment]
Thank you @cmwilson21  @mchammer01 @jurre !! Excited to get my first docs contribution :)

I'm happy to add this to the file, though it's substantially similar to the NuGet example. The ProGet user will likely know the endpoint url... but I don't think it hurts to add. Here's what I would add --- if it's okay, I can commit it .. or of course feel free to if it's easier for you :)

------------------------

Example of ProGet registry for Bundler (rubygems):

{% raw %}

```yaml
registries:
  proget-gems-feed:
    type: rubygems-server
    url: https://proget.corp.local/rubygems/MyRubygemsFeed
    token: ${{secrets.PROGET_APK_KEY}}
```

{% endraw %}

Example of ProGet registry for Python (PyPI):

{% raw %}

```yaml
registries:
  proget-python-feed:
    type: python-index
    url: https://proget.corp.local/pypi/MyPythonFeed
    token: ${{secrets.PROGET_APK_KEY}}
```

{% endraw %}

**Comment by mchammer01** [comment]
@whatatripp 👋🏻 - that's amazing, thanks for the quick reply ✨ 
Could you add these two examples to the files, I'll quickly review them from an editorial point of view, and @jurre will perform a technical review. Hopefully, we should be able to get this merged soon 😃

File: content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md (hunk 4 above)
Comment (whatatripp) [reply content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md:48]: Yes (Rubygems) - added an example  a bottom :)

File: content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md (hunk 5 above)
Comment (whatatripp) [reply content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md:355]: // same (Python feeds) - example added :)

**Comment by whatatripp** [comment]
@mchammer01 added :)

thanks for the guidance!

File: content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md [hunk 6]
```diff
@@ -627,3 +628,55 @@ If you are restricting which IPs can reach your Nexus host, you need to add the
 For more information, see [Securing Nexus Repository Manager](https://help.sonatype.com/repomanager3/planning-your-implementation/securing-nexus-repository-manager) in the Sonatype documentation.
 
   Registries can be proxied to reach out to a public registry in case a dependency is not available in the private regstry.{% ifversion dependabot-private-registries %} However, you may want {% data variables.product.prodname_dependabot %} to only access the private registry and not access the public regsitry at all. For more information, see [Quick Start Guide - Proxying Maven and NPM](https://help.sonatype.com/repomanager3/planning-your-implementation/quick-start-guide---proxying-maven-and-npm)" in the Sonatype documentation, and "[AUTOTITLE](/code-security/dependabot/working-with-dependabot/removing-dependabot-access-to-public-registries)."{% endif %}
+
+### ProGet
+
+For information about ProGet and instructions on how to configure {% data variables.product.prodname_dependabot %} to work with feeds in ProGet, see the [ProGet documentation](https://docs.inedo.com/docs/proget-overview).
+
+Example of ProGet registry for a NuGet feed:
```
Comment (mchammer01) [inline content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md:636]:
```suggestion
Example of ProGet registry configuration for a NuGet feed:
```

File: content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md [hunk 7]
```diff
@@ -627,3 +628,55 @@ If you are restricting which IPs can reach your Nexus host, you need to add the
...
+registries:
+  proget-nuget-feed:
+    type: nuget-feed
+    url: https://proget.corp.local/nuget/MyNuGetFeed/v3/index.json
+    token: ${{secrets.PROGET_APK_KEY}}
+```
+
+{% endraw %}
+
+Example of ProGet registry for Bundler (rubygems):
```
Comment (mchammer01) [inline content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md:650]:
```suggestion
Example of ProGet registry configuration for Bundler (rubygems):
```

File: content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md [hunk 8]
```diff
@@ -627,3 +628,55 @@ If you are restricting which IPs can reach your Nexus host, you need to add the
...
+registries:
+  proget-gems-feed:
+    type: rubygems-server
+    url: https://proget.corp.local/rubygems/MyRubygemsFeed
+    token: ${{secrets.PROGET_APK_KEY}}
+```
+
+{% endraw %}
+
+Example of ProGet registry for Python (PyPI):
```
Comment (mchammer01) [inline content/code-security/dependabot/working-with-dependabot/guidance-for-the-configuration-of-private-registries-for-dependabot.md:664]:
```suggestion
Example of ProGet registry configuration for Python (PyPI):
```

**Review by mchammer01 — COMMENTED** [review]
LGTM, just made some minor suggestions that I'm going to apply 😃

**Review by mchammer01 — APPROVED** [review]
Good to go from an editorial point of view!
@jurre - can you double-check that the examples of configurations that have been added are technically correct? Thanks 🙇🏻

**Review by jurre — APPROVED** [review]

**Pull request merged**